File: lock cracker.dart
void lock() {

    String lock = "821";

    for (var i = 0; i < 10; i++) {
        for (var j = 0; j < 10; j++) {
            for (var k = 0; k < 10; k++) {

                String i2 = i.toString();
                String j2 = j.toString();
                String k2 = k.toString();

                String output = i2 + j2 + k2;

                print(output);

                if (output == lock) {
                    print("lock combination found: " + output);
                    return;
                }
            }
        }
    }
    print("combination not found");
}

void main() {
    lock();
}

File: lock cracker.scala
object Main {

    def lock(): Unit = {

        var lock: String = "569"

        for (i <- 0 to 9) {
            for (j <- 0 to 9) {
                for (k <- 0 to 9) {

                    var i2: String = i.toString
                    var j2: String = j.toString
                    var k2: String = k.toString

                    var output: String = i2 + j2 + k2

                    println(output)

                    if (lock == output) {
                        println("lock combination found: " + output)
                        return
                    }
                }
            }
        }
        println("combination not found")
        return
    }
    
    def main(args: Array[String]): Unit = {
        lock()
    }
}
